### DISCUSSION TITLE
MPC controller that can handle multiple models for distributed control?

### DISCUSSION THREAD

**Discussion by Jushan-Randy-Chen** [do-mpc/do-mpc#325] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
I am trying to implement distributed control using do_mpc. Currently I am trying to run trajectory optimization for a swarm of quadrotors, and I realize that it is not so efficient with the current limitations of the package.

**Describe the solution you'd like**
Maybe we could have a `MPC.controller()` that can handle multiple model instances?

**Additional context**

**Comment by 4flixt** [comment, marked as answer]
Dear @RandyChen233, 

thank you for your question. Currently, do-mpc does not support the composition of models (e.g. different components of a chemical plant) to form a more complex model.
However, this wouldn't be necessary for distributed MPC. Considering your example: You would anyways require an individual MPC controller for each drone to enable distributed MPC. These controllers must then exchange information. 
This is absolutely possible with the current features provided by do-mpc: In particular, you would use the time-varying parameters.

Best,
4flixt
